src/iter/mod.rs

```diff
@@ -215,6 +217,13 @@ pub trait ParallelIterator: Sized + Send {
         flat_map::new(self, map_op)
     }
 
+    /// Reduces a collection of items into one collection. If you have
+    /// a `Vec<Vec<i32>>` it can be used to turn it into a `Vec<i32>`.
+    fn flatten(self) -> Flatten<Self>
```

Comment:
I'd like to see an example here. I find that is always super helpful to understand what's going on. e.g.,

```
use rayon::prelude::*;
let x: Vec<Vec<i32>> = vec![vec![1, 2], vec![3, 4]];
let y: Vec<i32> = x.into_par_iter().flatten().collect();
assert_eq!(y, vec![1, 2, 3, 4]);
```